main.tf

```
@@ -546,6 +546,416 @@ module "external_secrets" {
   tags = var.tags
 }
 
+################################################################################
+# AWS Load Balancer Controller
+################################################################################
+
+locals {
+  aws_load_balancer_controller_name = "aws-load-balancer-controller"
+  aws_load_balancer_controller_service_account = try(var.aws_load_balancer_controller.service_account_name, "${local.aws_load_balancer_controller_name}-sa")
+}
+
+data "aws_iam_policy_document" "aws_load_balancer_controller" {
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["iam:CreateServiceLinkedRole"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "iam:AWSServiceName"
+      values   = ["elasticloadbalancing.amazonaws.com"]
```

I don't think we had this before, but probably should have it
```suggestion
      values   = ["elasticloadbalancing.${data.aws_partition.current.dns_suffix}"]
```